### PR TITLE
Make test order deterministic for `pytest-xdist`

### PR DESCRIPTION
**Pull request by not522**
## Motivation
`pytest-xdist` is useful for parallelizing tests and reducing execution time, but Optuna's tests result in errors because the order is not deterministic. This PR fixes this error. After merging this PR, `pytest -m "not integration" -n auto` will succeed.

## Description of the changes
Make test order deterministic.

### CODE REVIEW

**Comment by not522** [comment]
@HideakiImamura @Alnusjaponica Could you review this PR?

**Comment by HideakiImamura** [comment]
Should we add `pytest-xdist` to `pyproject.toml` in `test`?

**Comment by HideakiImamura** [comment]
Thanks for the PR. I confirmed that we can execute `pytest -m "not integration" -n auto` with this PR.

**Review by Alnusjaponica — APPROVED** [review]
LGTM. It might be better to add some instruction in `optuna/blob/master/CONTRIBUTING.md`

File: tests/storages_tests/test_journal.py [hunk 1]
```diff
@@ -23,19 +23,14 @@
...
     "file_with_link_lock",
     "redis_default",
     "redis_with_use_cluster",
-}
+]
 
-LOG_STORAGE_SUPPORTING_SNAPSHOT = {
-    "redis_default",
-    "redis_with_use_cluster",
-}
```
Comment (not522) [inline tests/storages_tests/test_journal.py:36]: This variable has not been used.

File: tests/storages_tests/test_journal.py [hunk 2]
```diff
@@ -23,19 +23,14 @@
...
 
 
-LOG_STORAGE = {
+LOG_STORAGE = [
     "file_with_open_lock",
     "file_with_link_lock",
     "redis_default",
     "redis_with_use_cluster",
-}
+]
```
Comment (not522) [inline tests/storages_tests/test_journal.py:31]: The order of `set` items is not deterministic. As a result, `pytest-xdist` recognizes the collected tests are different between workers.

**Comment by not522** [comment]
> Should we add `pytest-xdist` to `pyproject.toml` in test?

`pytest-xdist` may give confusing errors, so I don't recommend adding it in CI. I think it is useful in local environments.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the reply. I understood. LGTM.

**Pull request merged**